src/notification.cpp

```diff
@@ -203,6 +207,55 @@ QVariantList decodeActionHints(const QHash<QString, QString> &actions, const QVa
 
 }
 
+NotificationData::NotificationData()
+    : replacesId(0)
+    , expireTimeout(-1)
+{
+}
+
+class NotificationPrivate : public NotificationData
+{
+    friend class Notification;
+
+    NotificationPrivate()
+        : NotificationData()
+    {
+    }
+
+    NotificationPrivate(const NotificationData &data)
+        : NotificationData(data)
+        , remoteActions(decodeActionHints(actions, hints))
+    {
+    }
+
+    NotificationPrivate &operator=(const NotificationData &data)
```

Comment:
It's as assignment from the wire data structure, not a normal assignment operator.